### ISSUE TITLE
AoT Compilation error: 

### ISSUE THREAD

**Issue by minuz** [udos86/ng2-dynamic-forms#267]
'DynamicFormService' is not exported by node_modules\@ng2-dynamic-forms\core\bundles\core.umd.js (imported by wwwroot\ng\customTypes\services\custom.type.service.js). For help fixing this error see https://github.com/rollup/rollup/wiki/Troubleshooting#name-is-not-exported-by-module
Error: 'DynamicFormService' is not exported by node_modules\@ng2-dynamic-forms\core\bundles\core.umd.js (imported by wwwroot\ng\customTypes\services\custom.type.service.js). For help fixing this error see https://github.com/rollup/rollup/wiki/Troubleshooting#name-is-not-exported-by-module

When running my project's AoT compilation, I get this error now.

Btw, the failure comes from rollup

**Comment by udos86** [comment]
@minuz Thanks for the issue! 

[I'm already aware](https://github.com/udos86/ng2-dynamic-forms/issues/257) that bundling the AoT and library UMD files with Rollup causes this error. I'm a bit confused because ng2 Dynamic Forms' UMD bundles are build with Rollup itself following the exact same structure than Angular 2 Material's bundle.

For the moment I would propose to you to use Webpack 2 and everything will work fine!

I'll have to do some further investigation here. Maybe you would like to help?

**Comment by minuz** [comment]
I'll see what I can do. I cannot switch from AoT to Webpack in my project. At least not for now. It is an ongoing project and I have a quite short deadline so I'll do some tests to fix this AoT problem.

**Comment by udos86** [comment]
@minuz Ok. The problem here is that Rollup itself [seems not to be](https://github.com/rollup/rollup-plugin-commonjs/issues/139) capable of handling UMD library files in the bundling process. 

It's a great tool for building UMD bundles but fails when resolving it while bundling. This is pretty weird.

**Comment by udos86** [comment]
@minuz I've got some great news for you! I've found a solution to make Rollup bundling work. I switched to `gulp-better-rollup` package for bundling my library files. For whatever reason the outcoming bundles will bundle with Rollup. I will release a new version at the end of the weekend!

**Comment by udos86** [comment]
@minuz Ok, I guess, I've now ultimately found the reason why bundling does not work in Rollup. It's the minified bundles! When using the unminified bundles everything seems to work fine. I'll try to deliver a package update that should work for Webpack as well as Rollup by the end of the weekend.

**Comment by udos86** [comment]
I'll deliver a fix for this problem by the end of the weekend

**Comment by minuz** [comment]
Hi @udos86,

As requested, I switched to development branch, built the modules and copied the packages across my project and ran my AoT.

Sadly, the change did not fix my rollup issue. I still receive the errors: 
``` javascript

{ code: 'MISSING_EXPORT',
  message: '\'DynamicFormService\' is not exported by \'node_modules\\@ng2-dynamic-forms\\core\\src\\service\\dynamic-form.service.js\'',
  url: 'https://github.com/rollup/rollup/wiki/Troubleshooting#name-is-not-exported-by-module',
  pos: 44608,
  loc:
   { file: 'C:\\GitHub\\Lumen.Workbench\\Lucsan.Workbench\\Lucsan.Workbench.WebApp\\wwwroot\\aot\\wwwroot\\ng\\app.module.ngfactory.js',
     line: 869,
     column: 61 },
  frame: '867:         get: function () {\r\n868:             if ((this.__DynamicFormService_87 == null)) {\r\n869:                 (this.__DynamicFormService_87 = new import60.DynamicFormService(this._FormBuilder_86, this.parent.get(import150.NG_VALIDATORS, null), this.parent.get(import150.NG_ASYNC_VALIDATORS, null)));\r\n                                                                  ^\n870:             }\r\n871:             return this.__DynamicFormService_87;\r',
  toString: [Function] }
{ code: 'MISSING_EXPORT',
  message: '\'DynamicFormAutoFillService\' is not exported by \'node_modules\\@ng2-dynamic-forms\\core\\src\\service\\dynamic-form-autofill.service.js\'',
  url: 'https://github.com/rollup/rollup/wiki/Troubleshooting#name-is-not-exported-by-module',
  pos: 45151,
  loc:
   { file: 'C:\\GitHub\\Lumen.Workbench\\Lucsan.Workbench\\Lucsan.Workbench.WebApp\\wwwroot\\aot\\wwwroot\\ng\\app.module.ngfactory.js',
     line: 879,
     column: 69 }
``` 

My rollup is as vanilla as possible. Based on Angular 2 cookbook and it is as follows:

``` javascript
import rollup from 'rollup'
import nodeResolve from 'rollup-plugin-node-resolve'
import commonjs from 'rollup-plugin-commonjs';
import uglify from 'rollup-plugin-uglify'

export default {
  entry: 'wwwroot/ng/main-aot.js',
  dest: 'wwwroot/dist/build.js', // output a single application bundle
  sourceMap: true,
  sourceMapFile: 'wwwroot/dist/build.js.map',
  format: 'iife',
  onwarn: function (warning) {
    if (warning.code === 'THIS_IS_UNDEFINED') { return; }
    console.error(warning);
  },
  plugins: [
    nodeResolve({ jsnext: true, module: true }),
    commonjs({
      include: [
        'node_modules/rxjs/**',
      ]
    }),
    uglify()
  ]
}
```

You mentioned earlier about the usage of the udm vs the udm.min where can you define which file it will use?

**Comment by udos86** [comment]
@minuz  I guess I've been on the wrong track! 

You could try one last thing. But this is really weird. It works for me but I don't know why.

Go to he node_modules folder in `@ng2-dynamic-forms`, delete the `umd.min.js` files and rename the `umd.js` files to some other name. 

That way I got my Rollup production bundling to work. But I've got now explanation why.

**Comment by minuz** [comment]
@udos86,  I just tried your suggestion with no joy. Still receive the same issue.

I'm unsure on what to do atm. I did a test copying your modules directly into my project and then it works fine. Sorry I can't be more helpful on this issue. I have no experience publishing libraries. I'll keep looking for an answer though.

**Comment by udos86** [comment]
@minuz Ok, here's another chance: Could you please adjust the `onwarn` function in your Rollup config as follows:

```
    onwarn: function (warning) {

        // https://github.com/rollup/rollup/wiki/Troubleshooting#this-is-undefined
        if (warning.code === "THIS_IS_UNDEFINED" || warning.code === "MISSING_EXPORT") {
            return;
        }

        console.error(warning);
    }
```

**Comment by gms1** [comment]
> 
> I'm a bit confused because ng2 Dynamic Forms' UMD bundles are build with Rollup itself following the exact same structure as Angular 2 Material's UMD bundle.

hm, The @angular / material package offers esm modules (ES5 version with ES2015 modules) and es5 UMD bundles. In package.json, the formers are referred by the "module" property, and the later is rerferred by the "main" property

The @ ng2-dynamic-forms packages only provides the "main" entry referring to the UMD bundle and the  commonjs modules (using require and having the __esModule property), typescript files, ....

**Comment by udos86** [comment]
@gms1 Now that's a nice hint! Thanks for this! I'll check this out!

**Comment by udos86** [comment]
@gms1 Wow, I guess, you probably found the missing puzzle piece. After adding `"module"` property to my `package.json` no missing import errors are thrown for ng2 Dynamic Forms any more. Thanks a lot for helping out!

**Comment by gms1** [comment]
you are welcome! Then it seems that the ESM modules are only required to make rollup's tree shaking happy

**Comment by minuz** [comment]
Hi @udos86, I just installed the new version and tried to run my AoT. Is this fix applied to version 1.3.17?

I'm still getting the same error :(
